### PR TITLE
 Refactor `wasmer_config::package::PackageHash` to be an enum

### PR DESCRIPTION
**Pull request by PappuPanaut**
# Description

 Code cleanup:
 wasmer_config::package::PackageHash - Refactor to enum #4585

### CODE REVIEW

File: lib/config/src/package/package_hash.rs [hunk 1]
```diff
@@ -4,29 +4,35 @@ use crate::{hash::Sha256Hash, package::PackageParseError};
 ///
 /// Currently only supports the format: `sha256:<hash>`.
 #[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
-pub struct PackageHash(Sha256Hash);
+pub enum PackageHash {
+    Sha256(Sha256Hash),
+}
 
 impl PackageHash {
     const STR_PREFIX: &'static str = "sha256:";
```
Comment (maminrayej) [inline lib/config/src/package/package_hash.rs:12]: This should probably be called `ShA256_STR_PREFIX`.

**Review by maminrayej — COMMENTED** [review]
Otherwise looks reasonable to me.

File: lib/config/src/package/package_hash.rs (hunk 1 above)
Comment (PappuPanaut) [reply lib/config/src/package/package_hash.rs:12]: Shouldn't enum variants be camelcase.

Comment (maminrayej) [reply lib/config/src/package/package_hash.rs:13]: I meant the string:
```
const STR_PREFIX: &'static str = "sha256:";
```
should be:
```
const SHA256_STR_PREFIX: &'static str = "sha256:";
```

Comment (maminrayej) [reply lib/config/src/package/package_hash.rs:13]: @PappuPanaut Please revert changes on the enum variant and instead fix the const str name.

**Review by maminrayej — APPROVED** [review]

**Comment by Arshia001** [comment]
@PappuPanaut @maminrayej I assume merging main into the PR branch would fix the CI failure.

**Comment by maminrayej** [comment]
>  I assume merging main into the PR branch would fix the CI failure.

Yes.

**Comment by maminrayej** [comment]
Thanks.

**Pull request merged**